File: common/protcol/c#/module/gate_call_clientmodule.cs
/*this module file is codegen by juggle for c#*/
using System;
using System.Collections;
using System.Collections.Generic;

namespace module
{
    public class gate_call_client : juggle.Imodule 
    {
        public gate_call_client()
        {
			module_name = "gate_call_client";
        }

        public delegate void connect_server_sucesshandle();
        public event connect_server_sucesshandle onconnect_server_sucess;
        public void connect_server_sucess(ArrayList _event)
        {
            if(onconnect_server_sucess != null)
            {
                onconnect_server_sucess();
            }
        }

        public delegate void ack_heartbeatshandle();
        public event ack_heartbeatshandle onack_heartbeats;
        public void ack_heartbeats(ArrayList _event)
        {
            if(onack_heartbeats != null)
            {
                onack_heartbeats();
            }
        }

        public delegate void call_clienthandle(String argv0, String argv1, ArrayList argv2);
        public event call_clienthandle oncall_client;
        public void call_client(ArrayList _event)
        {
            if(oncall_client != null)
            {
                var argv0 = ((String)_event[0]);
                var argv1 = ((String)_event[1]);
                var argv2 = ((ArrayList)_event[2]);
                oncall_client( argv0,  argv1,  argv2);
            }
        }

	}
}

File: common/protcol/js/module/gate_call_clientmodule.js
/*this module file is codegen by juggle for js*/
function gate_call_client_module(){
    eventobj.call(this);
    Imodule.call(this, "gate_call_client");

    this.ntf_uuid = function(argv0){
        this.call_event("ntf_uuid", [argv0]);
    }

    this.connect_gate_sucess = function(){
        this.call_event("connect_gate_sucess", []);
    }

    this.connect_hub_sucess = function(argv0){
        this.call_event("connect_hub_sucess", [argv0]);
    }

    this.ack_heartbeats = function(){
        this.call_event("ack_heartbeats", []);
    }

    this.call_client = function(argv0, argv1, argv2){
        this.call_event("call_client", [argv0, argv1, argv2]);
    }

}
(function(){
    var Super = function(){};
    Super.prototype = Imodule.prototype;
    gate_call_client_module.prototype = new Super();
})();
gate_call_client_module.prototype.constructor = gate_call_client_module;

